src/Microsoft.SqlTools.ServiceLayer/Scripting/ScriptingScriptOperation.cs

```diff
@@ -218,16 +218,40 @@ private SqlScriptPublishModel BuildPublishModel()
                     "Scripting object count {0}, objects: {1}",
                     selectedObjects.Count(),
                     string.Join(", ", selectedObjects)));
-
+            
+            string server = GetServerNameFromLiveInstance(this.Parameters.ConnectionString);
             string database = new SqlConnectionStringBuilder(this.Parameters.ConnectionString).InitialCatalog;
+
+
             foreach (ScriptingObject scriptingObject in selectedObjects)
             {
-                publishModel.SelectedObjects.Add(scriptingObject.ToUrn(database));
+                publishModel.SelectedObjects.Add(scriptingObject.ToUrn(server, database));
             }
 
             return publishModel;
         }
 
+        private string GetServerNameFromLiveInstance(string connectionString)
+        {
+            string serverName = string.Empty;
```

Comment:
provided you move the return into the using statement, which I think makes sense.

---
In reply to: [127088389](https://github.com/Microsoft/sqltoolsservice/pull/410#discussion_r127088389) [](ancestors = 127088389)